diff --git a/config.h b/config.h
index c19cfc7..e52a6da 100644
--- a/config.h
+++ b/config.h
@@ -13,7 +13,7 @@
 #endif
 
 /* define redis_fstat to fstat or fstat64() */
-#if defined(__APPLE__) && !defined(MAC_OS_X_VERSION_10_6)
+#if defined(__APPLE__) && !defined(MAC_OS_X_VERSION_10_4)
 #define redis_fstat fstat64
 #define redis_stat stat64
 #else
@@ -22,7 +22,7 @@
 #endif
 
 /* test for backtrace() */
-#if defined(__APPLE__) || defined(__linux__)
+#if defined(__APPLE__) && defined(MAC_OS_X_VERSION_10_5) || defined(__linux__)
 #define HAVE_BACKTRACE 1
 #endif
 
